.github/workflows/test.yml

```diff
@@ -61,6 +61,17 @@ jobs:
         ln -sf $(pwd) src/github.com/ovn-org/ovn-kubernetes
         GOPATH=$(pwd) gcov2lcov -infile gover.coverprofile -outfile coverage.lcov
 
+    - name: Build docker image
+      run: |
+        echo ${{ secrets.GITHUB_TOKEN }} | docker login docker.pkg.github.com -u $GITHUB_ACTOR --password-stdin
```

Comment:
@dave-tucker the job fails to push to docker because of the following message
```
denied: Resource not accessible by integration
##[error]Process completed with exit code 1.
```

it seems you can not use the GITHUB_TOKEN from a PR, the alternatives I saw is creating another token,
can you help me to unblock this?

Reply:
So first off, we have `docker.yml` for the image builds which is already set up to trigger on whether any of the files in the images are changed, we could probably adapt that to push to the registry which might save some time.

Secondly `GITHUB_TOKEN` only has `read` permissions to the package registry on forks so it won't work. Secrets aren't available to forks either. Additionally, as the tag you are using isn't unique we can end up with non-deterministic test results since multiple jobs can push to the registry in parallel.

The best way to achieve what you want is to use artifacts a little like this:

```
- name: Build docker image
  run: |
    pushd dist/images
    sudo cp -f ../../go-controller/_output/go/bin/ovn* .
    echo "ref: $(git rev-parse  --symbolic-full-name HEAD)  commit: $(git rev-parse  HEAD)" > git_info
    docker build -t docker.pkg.github.com/${GITHUB_REPOSITORY}/ovn-daemonset-f:dev -f Dockerfile.fedora .
    mkdir _output
    docker save docker.pkg.github.com/${GITHUB_REPOSITORY}/ovn-daemonset-f:dev > _output/image.tar

- uses: actions/upload-artifact@v2
  with:
    name: test-image
    path: dist/images/_output/image.tar
```

And then later in the next job:

```
- uses: actions/download-artifact@v2
  with:
    name: test-image
- name: Load docker image
   run: |
      docker load --input image.tar
```

Reply:
awesome, it works perfect